lib/immutable-struct.rb

```diff
@@ -1,3 +1,5 @@
+require 'json'
```

Comment:
I worry this might break someone trying to use another JSON library than the Ruby standard.  I don't 100% know how those work, but I'm wondering if we can omit this `require`.  If someone is going to call `to_json`, they will have had to require some sort of JSON library elsewhere.